src/plugins/data/public/query/query_string/language_service/lib/query_result.tsx

```diff
@@ -28,21 +28,66 @@ export interface QueryStatus {
     statusCode?: number;
   };
   elapsedMs?: number;
+  startTime?: number;
 }
 
+// This is the time in milliseconds that the query will wait before showing the loading spinner
+const BUFFER_TIME = 3000;
+
 export function QueryResult(props: { queryStatus: QueryStatus }) {
   const [isPopoverOpen, setPopover] = useState(false);
+  const [elapsedTime, setElapsedTime] = useState(0);
   const onButtonClick = () => {
     setPopover(!isPopoverOpen);
   };
 
+  const getTime = () => {
+    const time = new Date().getTime() - props.queryStatus.startTime!;
+    if (time > BUFFER_TIME) {
+      setElapsedTime(time);
```

Comment:
Based on the interval below and the logic here, elapsedTime will be
0 for first run
0 for second run
3000 for third run
4000 for fourth run

Is that expected?

Reply:
i think 1s interval is a reasonable time to get elapsedTime here? since @ashwin-pc also requests we show loading time in seconds

Reply:
Its rounded down to seconds so that should not be a problem. Take a look at the video attached Miki :)